src/components/Text.tsx

```diff
@@ -0,0 +1,206 @@
+      if(typeof word !== "string")
+      {
+        let currentGreekWordNotes : GreekWordNotes = {} as GreekWordNotes;
+        let currentGreekWordAttributes: GreekWordAttributes;
+        let tempGreekWordNotes : any = {};
+
+        if(word.note !== undefined)
+        {
+          word.note.forEach((e) => {
+            let greekWordNoteKey : ValidGreekWordNoteKeys = e.ATTR.type;
+            let greekWordNoteValue : string = e._;
+            tempGreekWordNotes[greekWordNoteKey.slice(2)] = greekWordNoteValue;
+          });
+
+          currentGreekWordNotes = tempGreekWordNotes as GreekWordNotes;
+        }
+
+        currentGreekWordAttributes = word.ATTR;
+
+        let currentGreekWord : GreekWord= {notes : currentGreekWordNotes, attributes : currentGreekWordAttributes, text: word._}
+
+        // if phraseWord or subWord, insert to buffer
+        if(currentGreekWordNotes.phraseWords !== undefined || currentGreekWordNotes.sub !== undefined) 
+        {
+          hasConsumedPhraseWord = (currentGreekWordNotes.phraseWords !== undefined );
+          hasConsumedSubWord = (currentGreekWordNotes.sub !== undefined);
+          greekWordBuffer.push(currentGreekWord);
+          currentPhraseWords = (hasConsumedPhraseWord && currentGreekWordNotes.phraseWords !== undefined) ? currentGreekWordNotes.phraseWords : "";
+        }
+        else // is a normal greek word
+        {
+          let greekWords : GreekWord[] = [];
+
+          // check if buffer needs to be emptied. 
+          if(hasConsumedPhraseWord || hasConsumedSubWord)
+          {
+            // NOTE: for some reason, this loop prevents the data being passed to the onPhraseClick from being overwritten
+            let length = greekWordBuffer.length;
+            for(let j = 0; j < length; j++)
+            {
+              let tempGreekWord = greekWordBuffer.pop() as GreekWord;
+              greekWords.unshift(tempGreekWord);
+            }
+          }
+          
+          if(hasConsumedPhraseWord) // if buffer contains phrase words NOTE: current word is NOT processed
+          {
+            verseWordOutput.push(<><span className="GreekPhrase" onClick={() => {onPhraseClick([...greekWords])}}>{currentPhraseWords}</span><span> </span></>)
+            hasConsumedPhraseWord = false;
+            greekWordBuffer.splice(0, greekWordBuffer.length);
+          }
+          
+          if(hasConsumedSubWord)// buffer contain subWords. NOTE: Current word is processed in this case
+          {
+            currentGreekWord.text = currentGreekWord.text.replace('[1]', greekWords[0].text);
+            greekWords.push(currentGreekWord)
+            verseWordOutput.push(<><span className="GreekPhrase" onClick={() => {onPhraseClick([...greekWords])}}>{currentGreekWord.text}</span><span> </span></>)
+            hasConsumedSubWord = false;
+            greekWordBuffer.splice(0, greekWordBuffer.length);
+          }
+          else if(word._ !== "√") // current word has english backing
```

Comment:
What is the underscore in this context?  It looks like a string that contains a checkmark unicode character?  Could this be a boolean?

Reply:
Symbol indicates that the word has no English meaning